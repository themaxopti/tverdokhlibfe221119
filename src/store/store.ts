import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import usersReducer from "./reducers/users.slice";

export const store = configureStore({
  reducer: {
    users: usersReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
