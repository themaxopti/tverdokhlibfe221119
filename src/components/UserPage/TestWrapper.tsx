import { Provider } from "react-redux";
import { store } from "../../store/store";
import { MemoryRouter } from "react-router-dom";
import { Router } from "../../utils/Router/Router";

export const TestWrapper = ({ initialRoute }: any) => {
    return (
        <Provider store={store}>
            <MemoryRouter initialEntries={[initialRoute]}><Router /></MemoryRouter>
        </Provider>
    );
};