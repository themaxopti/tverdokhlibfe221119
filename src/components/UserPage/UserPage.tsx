import React, { useState, useEffect } from "react";
import hideImg from "../../assets/images/close.png";
import { Link } from "react-router-dom";
import { useSelector } from "react-redux";
import { useAppSelector } from "../../store/hooks";
import { useParams } from "react-router-dom";
import { User, User as UserType } from "../../store/reducers/users.slice";


interface UserPageProps {
    currentUser: User | undefined
}

export const UserPage: React.FC<UserPageProps> = ({ currentUser }) => {


    return (
        <div data-testid="userPage" className="user-page">
            <div className="user-page__hide">
                <Link to={"/"}>
                    <img src={hideImg} alt="" />
                </Link>
            </div>
            <div className="user-page__content">
                <div className="user-page__about">
                    <div className="user-page__photo">
                        {currentUser && <img src={`/images/${currentUser.id}.jpg`} alt="" />}
                    </div>
                    <div className="user-page__surname">{currentUser?.name}</div>
                    <div className="user-page__position">{currentUser?.position}</div>
                </div>
                <div className="user-page__contacts">
                    <div className="user-page__contact">
                        <div className="user-page__title">Phone</div>
                        <div>{currentUser?.phone}</div>
                    </div>
                    <div className="user-page__contact">
                        <div className="user-page__title">NickName</div>
                        <div>{currentUser?.nickname}</div>
                    </div>
                    <div className="user-page__contact">
                        <div className="user-page__title">Email</div>
                        <div>{currentUser?.email}</div>
                    </div>
                </div>
                <div className="user-page__send">Send Message</div>
            </div>
        </div>
    );
};
