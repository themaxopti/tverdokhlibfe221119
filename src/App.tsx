import React from "react";
import logo from "./logo.svg";
import { Users } from "./components/Users/Users";
import { User } from "./components/User/User";
import { UserPage } from "./components/UserPage/UserPage";
import { Link, Routes, Route } from "react-router-dom";
import { Router } from "./utils/Router/Router";

function App() {
  return (
    <div className="App">
      <Router />
    </div>
  );
}

export default App;
