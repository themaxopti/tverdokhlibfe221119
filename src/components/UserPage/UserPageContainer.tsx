import React, { useState, useEffect } from "react";
import { useParams } from "react-router-dom";
import { useAppSelector } from "../../store/hooks";
import { User as UserType } from "../../store/reducers/users.slice";
import { UserPage } from './UserPage';

export const UserPageContainer = () => {

    const [currentUser, setCurrentUser] = useState<UserType | undefined>(undefined);
    const users = useAppSelector((state) => state.users.users);
    let { id } = useParams();

    useEffect(() => {
        const user = users.find((user) => String(user.id) === id);
        setCurrentUser(user);
    }, []);
    return (
        <UserPage currentUser={currentUser} />
    )
}
