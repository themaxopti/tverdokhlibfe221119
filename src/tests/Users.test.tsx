import React from "react";
import { fireEvent, getByText, render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "../store/store";
import App from "../App";
import { TestWrapper } from "../components/UserPage/TestWrapper";

test("renders learn react link", () => {
  const Users = render(
    <TestWrapper initialRoute={'/'} />
  );
  const users = screen.getByTestId("users")
  const user = screen.getAllByTestId("user")
  expect(users).toBeInTheDocument();
  expect(user.length).toBe(3)
  const viewBtn = screen.getByTestId('view')
  fireEvent.click(viewBtn)
  const newUsers = screen.getAllByTestId("user")
  expect(newUsers.length).toBe(6)
  expect(Users).toMatchSnapshot()
});
