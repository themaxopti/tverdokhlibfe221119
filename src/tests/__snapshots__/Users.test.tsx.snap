// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="users"
        data-testid="users"
      >
        <div
          class="users__items"
        >
          <div
            class="users__item"
            data-testid="user"
          >
            <div
              class="users__information"
            >
              <div
                class="users__image"
              >
                <img
                  alt=""
                  src="/images/1.jpg"
                />
              </div>
              <div
                class="users__name"
              >
                 
                Geordan Aaryn
              </div>
              <div
                class="users__nickname "
              >
                @geordanaaryn
              </div>
            </div>
            <a
              href="/user/1"
            >
              <div
                class="users__view"
              >
                View
              </div>
            </a>
          </div>
          <div
            class="users__item"
            data-testid="user"
          >
            <div
              class="users__information"
            >
              <div
                class="users__image"
              >
                <img
                  alt=""
                  src="/images/2.jpg"
                />
              </div>
              <div
                class="users__name"
              >
                 
                Brazil Izair
              </div>
              <div
                class="users__nickname "
              >
                @brazilizair
              </div>
            </div>
            <a
              href="/user/2"
            >
              <div
                class="users__view"
              >
                View
              </div>
            </a>
          </div>
          <div
            class="users__item"
            data-testid="user"
          >
            <div
              class="users__information"
            >
              <div
                class="users__image"
              >
                <img
                  alt=""
                  src="/images/3.jpg"
                />
              </div>
              <div
                class="users__name"
              >
                 
                Selman Winston
              </div>
              <div
                class="users__nickname "
              >
                @selmanwinston
              </div>
            </div>
            <a
              href="/user/3"
            >
              <div
                class="users__view"
              >
                View
              </div>
            </a>
          </div>
          <div
            class="users__item"
            data-testid="user"
          >
            <div
              class="users__information"
            >
              <div
                class="users__image"
              >
                <img
                  alt=""
                  src="/images/4.jpg"
                />
              </div>
              <div
                class="users__name"
              >
                 
                Kaine Dutch
              </div>
              <div
                class="users__nickname "
              >
                @kainedutch
              </div>
            </div>
            <a
              href="/user/4"
            >
              <div
                class="users__view"
              >
                View
              </div>
            </a>
          </div>
          <div
            class="users__item"
            data-testid="user"
          >
            <div
              class="users__information"
            >
              <div
                class="users__image"
              >
                <img
                  alt=""
                  src="/images/5.jpg"
                />
              </div>
              <div
                class="users__name"
              >
                 
                Kerby Linton
              </div>
              <div
                class="users__nickname "
              >
                @kerbylinton
              </div>
            </div>
            <a
              href="/user/5"
            >
              <div
                class="users__view"
              >
                View
              </div>
            </a>
          </div>
          <div
            class="users__item"
            data-testid="user"
          >
            <div
              class="users__information"
            >
              <div
                class="users__image"
              >
                <img
                  alt=""
                  src="/images/6.jpg"
                />
              </div>
              <div
                class="users__name"
              >
                 
                Gilad Delvon
              </div>
              <div
                class="users__nickname "
              >
                @giladdelvon
              </div>
            </div>
            <a
              href="/user/6"
            >
              <div
                class="users__view"
              >
                View
              </div>
            </a>
          </div>
        </div>
        <div
          class="users__all"
          data-testid="view"
        >
          View all
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="users"
      data-testid="users"
    >
      <div
        class="users__items"
      >
        <div
          class="users__item"
          data-testid="user"
        >
          <div
            class="users__information"
          >
            <div
              class="users__image"
            >
              <img
                alt=""
                src="/images/1.jpg"
              />
            </div>
            <div
              class="users__name"
            >
               
              Geordan Aaryn
            </div>
            <div
              class="users__nickname "
            >
              @geordanaaryn
            </div>
          </div>
          <a
            href="/user/1"
          >
            <div
              class="users__view"
            >
              View
            </div>
          </a>
        </div>
        <div
          class="users__item"
          data-testid="user"
        >
          <div
            class="users__information"
          >
            <div
              class="users__image"
            >
              <img
                alt=""
                src="/images/2.jpg"
              />
            </div>
            <div
              class="users__name"
            >
               
              Brazil Izair
            </div>
            <div
              class="users__nickname "
            >
              @brazilizair
            </div>
          </div>
          <a
            href="/user/2"
          >
            <div
              class="users__view"
            >
              View
            </div>
          </a>
        </div>
        <div
          class="users__item"
          data-testid="user"
        >
          <div
            class="users__information"
          >
            <div
              class="users__image"
            >
              <img
                alt=""
                src="/images/3.jpg"
              />
            </div>
            <div
              class="users__name"
            >
               
              Selman Winston
            </div>
            <div
              class="users__nickname "
            >
              @selmanwinston
            </div>
          </div>
          <a
            href="/user/3"
          >
            <div
              class="users__view"
            >
              View
            </div>
          </a>
        </div>
        <div
          class="users__item"
          data-testid="user"
        >
          <div
            class="users__information"
          >
            <div
              class="users__image"
            >
              <img
                alt=""
                src="/images/4.jpg"
              />
            </div>
            <div
              class="users__name"
            >
               
              Kaine Dutch
            </div>
            <div
              class="users__nickname "
            >
              @kainedutch
            </div>
          </div>
          <a
            href="/user/4"
          >
            <div
              class="users__view"
            >
              View
            </div>
          </a>
        </div>
        <div
          class="users__item"
          data-testid="user"
        >
          <div
            class="users__information"
          >
            <div
              class="users__image"
            >
              <img
                alt=""
                src="/images/5.jpg"
              />
            </div>
            <div
              class="users__name"
            >
               
              Kerby Linton
            </div>
            <div
              class="users__nickname "
            >
              @kerbylinton
            </div>
          </div>
          <a
            href="/user/5"
          >
            <div
              class="users__view"
            >
              View
            </div>
          </a>
        </div>
        <div
          class="users__item"
          data-testid="user"
        >
          <div
            class="users__information"
          >
            <div
              class="users__image"
            >
              <img
                alt=""
                src="/images/6.jpg"
              />
            </div>
            <div
              class="users__name"
            >
               
              Gilad Delvon
            </div>
            <div
              class="users__nickname "
            >
              @giladdelvon
            </div>
          </div>
          <a
            href="/user/6"
          >
            <div
              class="users__view"
            >
              View
            </div>
          </a>
        </div>
      </div>
      <div
        class="users__all"
        data-testid="view"
      >
        View all
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
